pkg/services/ngalert/store/alert_rule.go

```diff
@@ -114,7 +114,22 @@ func (st DBstore) GetAlertRulesGroupByRuleUID(ctx context.Context, query *ngmode
 		if err != nil {
 			return err
 		}
-		result = rules
+		// make sure we keep the case-sensitive comparison.
```

Comment:
```suggestion
		// MySQL by default compares strings without case-sensitivity, make sure we keep the case-sensitive comparison.
```
nit. i can see someone reading this 1yr from now and wondering why